src/aap_eda/tasks/activation_request_queue.py

```diff
@@ -12,19 +12,37 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+from django.db import transaction
 from django.db.models.query import QuerySet
+from django.db.utils import IntegrityError
 
-from aap_eda.core.enums import ActivationRequest
-from aap_eda.core.models import ActivationRequestQueue
+from aap_eda.core.enums import ActivationRequest, ProcessParentType
+from aap_eda.core.models import Activation, ActivationRequestQueue, EventStream
 
+from .exceptions import UnknownProcessParentType
 
+
+@transaction.atomic
 def push(parent_type: str, parent_id: int, request: ActivationRequest) -> None:
+    if parent_type == ProcessParentType.ACTIVATION:
+        model = Activation
+    elif parent_type == ProcessParentType.EVENT_STREAM:
+        model = EventStream
+    else:
+        raise UnknownProcessParentType(
+            f"Unknown parent type {parent_type}",
+        )
+
     ActivationRequestQueue.objects.create(
         process_parent_type=parent_type,
         process_parent_id=parent_id,
         request=request,
     )
 
+    # Check that the parent referenced still exists.
```

Comment:
Shouldn't we do this before creating the registry?

Reply:
Doesn't matter much as we have two potential asynchronous operations both manipulating the db; this enqueue (on the default worker) and the actual delete (on an activation worker).  Doing it as late as possible narrows the chances of encountering issues from that.

If we really want the whole thing to be bullet-proof w/o a complicated synchronization approach we should modify the handling of auto-start handling to consider the determination the activation no longer exists to be dispositive and drop the auto-start from the requests queue.

Reply:
This is not exclusively related with the autostart. This is to recover a functionality that we already had and we lost [with this change](https://github.com/ansible/eda-server/pull/656/files#diff-71687df2e87df81216aba65e07df0f26fd2fca9fc6fa2eb1aaa275db0319b5cbR47). 
Although cancelling the autostart job might be good, the real issue is caused because we are creating the record when we should not do it because the process_parent does not exist. Prior to this change, this was guaranteed due to the foreign key, which is why we caught the exception in _queue_auto_start. The activation no longer exists, so there is no need to autostart anything.

With this change, we miss the constraint and we have to take care of it explicitly (that's because it was decided to implement the relationship "manually" instead of using Django [generic relations](https://docs.djangoproject.com/en/5.0/ref/contrib/contenttypes/#generic-relations)). It could be the source of more bugs, but I think we already take care of non-existing process_parents. For 100% consistency, we should take care of the on_delete_cascade; we can do that in a different PR.

But what we cannot do is raise an IntegrityError after creating the object.
IntegrityError means the opposite: the record cannot be created because the related object does not exist, and we have to raise the exception before creating the object.

Reply:
I missed the atomic decorator.
all is ok then. :)